### PR TITLE
Add switch to 3D building #7

### PR DESCRIPTION
**Pull request by frodrigo**
First type script ever.

The 3d building layer is not yet deployed.

Peek 22-05-2020 12-25

### CODE REVIEW

File: docs/building3d/index.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      });
+
+      map.on('load', function() {
+        const building = teritorio.building3d(map);
+
+        document.querySelector('.top-buttons').addEventListener('click', function(event) {
+          event.stopPropagation();
+
+          if (event.target.dataset.is3d) {
+            building.set3d(event.target.dataset.is3d == 'true');
```
Comment (GuillaumeAmat) [inline docs/building3d/index.html:40]:
```suggestion
            building.set3d(event.target.dataset.is3d === 'true');
```

File: src/building3d.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import mapboxgl from 'mapbox-gl';
+
+interface Options {
+  defaut3d?: boolean;
+  defaultPitch?: any;
```
Comment (GuillaumeAmat) [inline src/building3d.ts:5]:
```suggestion
  defautl3d?: boolean;
  defaultPitch?: number;
```

File: src/building3d.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+/**
+ * Select and switch 3D building.
+ */
+export function building3d(map: mapboxgl.Map, options?: Options) {
+  if (options?.defaut3d) {
```
Comment (GuillaumeAmat) [inline src/building3d.ts:12]:
```suggestion
  if (options?.default3d) {
```

File: src/building3d.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    set3d(map, options?.defaut3d, options?.defaultPitch);
```
Comment (GuillaumeAmat) [inline src/building3d.ts:13]:
```suggestion
    set3d(map, options?.default3d, options?.defaultPitch);
```

File: src/building3d.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  return {
+    set3d(is3d: boolean, pitch: any) {
```
Comment (GuillaumeAmat) [inline src/building3d.ts:17]:
```suggestion
    set3d(is3d: boolean, pitch: number) {
```

File: src/building3d.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      set3d(map, is3d, pitch);
+    },
+  };
+}
+
+export function set3d(map: mapboxgl.Map, is3d: boolean, pitch: any) {
```
Comment (GuillaumeAmat) [inline src/building3d.ts:23]:
```suggestion
export function set3d(map: mapboxgl.Map, is3d: boolean, pitch: number) {
```

File: src/building3d.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  if (is3d) {
+    // Make it 3D
+    if (pitch !== false) {
```
Comment (GuillaumeAmat) [inline src/building3d.ts:26]:
```suggestion
    if (typeof pitch !== 'undefined') {
```

File: test/building3d.test.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import { icon } from '../src';
```
Comment (GuillaumeAmat) [inline test/building3d.test.ts:1]:
```suggestion
import { building3d } from '../src';
```

File: test/building3d.test.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+
+describe('building3d', () => {
+  test('is exported', () => {
+    expect(icon).toBeTruthy();
```
Comment (GuillaumeAmat) [inline test/building3d.test.ts:5]:
```suggestion
    expect(building3d).toBeTruthy();
```

**Comment by GuillaumeAmat** [comment]
Just a few things, and welcome to TypeScript! ;)

For the record, we should always avoid using `any` to type things. I had to use it for the `icon`  function because I wanted to use the same types as in Mapbox GL, but it's an exception.

**Comment by frodrigo** [comment]
About `pitch` and `defaultPitch` my goal was to have 60 as default with the ability to disable the `pitch` by setting it to false. Maybe not the good way to do this.

File: src/building3d.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      map.easeTo({ pitch: pitch || 60, duration: 500 });
```
Comment (GuillaumeAmat) [inline src/building3d.ts:27]:
```suggestion
      map.easeTo({ pitch, duration: 500 });
```

**Comment by GuillaumeAmat** [comment]
> About `pitch` and `defaultPitch` my goal was to have 60 as default with the ability to disable the `pitch` by setting it to false. Maybe not the good way to do this.

You're right, I forgot about that, I updated the suggestions to do that in an other way.

**Comment by frodrigo** [comment]
All fixed and commit squashed.
I let you merge if it is now ok.

File: src/building3d.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import mapboxgl from 'mapbox-gl';
+
+interface Options {
+  defaut3d?: boolean;
```
Comment (GuillaumeAmat) [inline src/building3d.ts:4]: There is still this one to fix @frodrigo

**Pull request merged**